### ISSUE TITLE
Datablock fails to parse with obtuse error when images missing

### ISSUE THREAD

**Issue by ndevenish** [dials/dials#503]
```
dials.find_spots datablock.json min_spot_size=1
Sorry: Unable to handle the following arguments:
  datablock.json failed during DataBlock processing:
    Traceback (most recent call last):
      File "/Users/rze73714/dials/modules/dials/util/options.py", line 310, in try_read_datablocks
        self.datablocks.append(converter.from_string(argument))
      File "/Users/rze73714/dials/modules/dials/util/phil.py", line 47, in from_string
        check_format=self._check_format))
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 945, in from_json_file
        return DataBlockFactory.from_json(infile.read(), check_format)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 935, in from_json
        check_format)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 923, in from_dict
        importer = DataBlockDictImporter(obj, check_format)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 660, in __init__
        self.datablocks = self._load_datablocks(obj, check_format)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 675, in _load_datablocks
        return [self._load_datablocks(dd, check_format) for dd in obj]
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/datablock.py", line 787, in _load_datablocks
        filenames, None, check_format, indices, format_kwargs=format_kwargs)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/imageset.py", line 858, in make_imageset
        format_class = Registry.find(filenames[0])
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/format/Registry.py", line 74, in find
        if format.understand(image_file):
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/format/FormatCBF.py", line 25, in understand
        if '###CBF' in FormatCBF.open_file(image_file, 'rb').read(6):
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/format/Format.py", line 601, in open_file
        .check(filename, fh_func)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/filecache_controller.py", line 63, in check
        return self._cache.open()
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/filecache.py", line 183, in open
        return pseudo_file(self)
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/filecache.py", line 326, in __init__
        self._cache_object.register()
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/filecache.py", line 208, in register
        self._check_not_closed()
      File "/Users/rze73714/dials/modules/cctbx_project/dxtbx/filecache.py", line 173, in _check_not_closed
        raise IOError('Accessing lazy file cache %s after closing is not allowed' % str(self))
    IOError: Accessing lazy file cache <dxtbx.filecache.lazy_file_cache instance at 0x1075c2c68> after closing is not allowed
```

Was remote report; my HDF5 example does:

```
Sorry: Unable to handle the following arguments:
  "datablock.json" failed during DataBlock processing:
    Traceback (most recent call last):
      File "/Users/nickd/dials/dist/modules/dials/util/options.py", line 319, in try_read_datablocks
        self.datablocks.append(converter.from_string(argument))
      File "/Users/nickd/dials/dist/modules/dials/util/phil.py", line 48, in from_string
        check_format=self._check_format))
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 991, in from_json_file
        return DataBlockFactory.from_json(infile.read(), check_format)
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 981, in from_json
        check_format)
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 969, in from_dict
        importer = DataBlockDictImporter(obj, check_format)
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 673, in __init__
        self.datablocks = self._load_datablocks(obj, check_format)
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 688, in _load_datablocks
        return [self._load_datablocks(dd, check_format) for dd in obj]
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/datablock.py", line 786, in _load_datablocks
        format_kwargs = format_kwargs)
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/imageset.py", line 525, in make_sweep
        format_class = Registry.find(filenames[0])
      File "/Users/nickd/dials/dist/modules/cctbx_project/dxtbx/format/Registry.py", line 83, in find
        raise IOError('no format support found for %s' % image_file)
    IOError: no format support found for /private/var/folders/54/rx9zwmy96sqbczfvw51plf900000gn/T/tmp.YskQtwds/transthyretin_h5/200Hz_0.1dpf_1_master.h5
```

Related: #495 appears to be the same underlying problem, #335 possibly.

**Comment by ndevenish** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Selina's bug - missing image files gives nonsensical errors](https://trello.com/c/10yrkShU/23-selinas-bug-missing-image-files-gives-nonsensical-errors)

**Comment by ndevenish** [comment]
Fixed with cctbx/cctbx_project@0970a2d3809567dac4e42c5fea68ffd42ae96a40, although the underlying issues in #335 remain. Now looks like:
```
Sorry: Unable to handle the following arguments:
  "datablock.json" failed during DataBlock processing:
    No such file or directory: /private/var/folders/54/rx9zwmy96sqbczfvw51plf900000gn/T/tmp.dtsVsOul/data/transthyretin_h5/200Hz_0.1dpf_1_master.h5
```
